opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -241,17 +280,49 @@ public long timestampMicros() {
         } else {
             this.tags = new HashMap<>(initialTags);
         }
+        if(refs == null) {
+            this.references = Collections.emptyList();
+        } else {
+            this.references = new ArrayList<>(refs);
+        }
+        MockContext parent = findPreferredParentRef(this.references);
         if (parent == null) {
             // We're a root Span.
             this.context = new MockContext(nextId(), nextId(), new HashMap<String, String>());
             this.parentId = 0;
         } else {
             // We're a child Span.
-            this.context = new MockContext(parent.traceId, nextId(), parent.baggage);
+            this.context = new MockContext(parent.traceId, nextId(), mergeBaggages(this.references));
             this.parentId = parent.spanId;
         }
     }
 
+    private static MockContext findPreferredParentRef(List<Reference> references) {
+        if(references.isEmpty()) {
+            return null;
+        }
+        Reference preferredReference = references.get(0);
+        for (Reference reference : references) {
+            // childOf takes precedence as a preferred parent
+            if (References.CHILD_OF.equals(reference.getReferenceType())
```

Comment:
why not just return immediately in this case?

Reply:
Indeed, I don't have a strong opinion on that. The control flow is the same. I find it more readable to have a single return at the end but I will change this.

Reply:
what I meant was this:
```
if list.isEmpty { return nil; }
for r in list {
  if r.isChildOf { return r.context() }
}
return list.get(0).context();
```
faster and simpler